### ISSUE TITLE
[Admin Tools] Media Manager

### ISSUE THREAD

**Issue by TheOneTrueLX** [TheOneTrueLX/OtakuDB-Issues#14]
# Story Statement
As an administrator, I want a media management front end so that I can better manage the image and video assets in use on the site, as well as manage links to off-site media.

# Supporting Information

**Is your feature request related to a problem? Please describe.**
This is new functionality not related to a current problem, although the lack of a media management solution would likely result in problems down the road.

**Describe the solution you'd like**
The media manager should provide the following features:

- The ability to list media currently uploaded to the system [already done]
- The ability to upload new media in bulk  [already done]
- The ability to automatically detect duplicate media and behave "appropriately"
- The ability to automagically generate thumbnails for uploaded media
- The ability to edit and remove media, and also view where that media is in use
- The ability to prune orphaned image and video files

**Describe alternatives you've considered**
There are 3rd party media managers available for Laravel, but none of them appear to be capable of being tailored to accommodate the unique needs of the site.  For the amount of effort that would be required to tailor one of these solutions to fit with ODB, creating a home-brewed solution that contains all of (and just) the features we need makes more sense.

**Additional context**
N/A

**Comment by TheOneTrueLX** [comment]
TL;DR: trying to track three separate types of "media" (images, videos, and external URLs to the same) is a dumb idea - need to refactor the media system to track each of these types of media separately.

**Comment by TheOneTrueLX** [comment]
Executive decision made not to do anything special for managing URLs at this time.  And with that, the media manager is for better or for worse completed.  Will need to revisit once we get some content records in place (since we're going to need to manage relationships that don't yet exist), but there's an issue to track that (#24 to be precise)
